### ISSUE TITLE
meerkat pre-processing step fail to generate insertion size file 

### ISSUE THREAD

**Issue by sckinta** [parklab/Software-Issues#20]
Please provide us with the following information so we can help you out:

#### The Software:

meerkat pre-process.pl

#### The Problem:
no insertion size information (prefix/prefix.is) were generated.
#### Details:
The pre-process.pl script seems successfully run. Please see log

> $ cat cartilage_P18S_loc1.srt.dupmark.pre.log
> warning: cannot open reagroup blacklist 'null'
> Options:
>    input BAM: /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.bam
>    prefix: /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark
>    generating blacklist: coverage >= 1500 reads
>    max insert size: 1000 bases
>    soft clip threshold: 60 bases
>    fragment size: 35 bases
>    insert size samples: 0 reads
>    blacklisted read groups: 
>    trim reads: 15
>    max N bps per read: 5
>    output files: 
>       /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.unmapped.fq.gz
>       /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.softclips.fq.gz
>       /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.isinfo
>       /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.unmapped.rdist
>       /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.softclips.rdist
>       /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.sr.1.fq.gz
>       /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.sr.2.fq.gz
>       /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.blacklist.gz
>       /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark/ (directory)
> Forcing immediate execution.
> /mnt/isilon/sfgi/suc1/analyses/grant/dnaSeq/HME/meerkat/cartilage_P18S_loc1/cartilage_P18S_loc1.srt.dupmark.bam: 
>    1178625399 reads
>    1877040 unaligned
>    54744079 clipped (42308175 hashed)
>    151 bps in the longest read
>    Quality string info:
>       quality score range: [35, 70]
>     * Sanger                   [33, 126]
>       Solexa/Illumina 1.0      [59, 126]
>       Illumina 1.3             [64, 126]
>       Illumina 1.5             [66, 126]
> Final read counts after rejection due to quality (-q 15 -m 120):
>    unmapped reads: 1543650 (333390 rejected)
>    clipped reads:  54545066 (199013 rejected)
> Readgroup [cartilage_P18S_loc1]: 1178625399 reads

However, the insertion size file is empty and insertion size plotting failed in R
> ls -lhtr cartilage_P18S_loc1.srt.dupmark 
> total 1.8G
> -rw-r--r-- 1 suc1 sfgi 679M Jan 29 12:53 cartilage_P18S_loc1_1.fq.gz
> -rw-r--r-- 1 suc1 sfgi 834M Jan 29 12:53 cartilage_P18S_loc1_2.fq.gz
> -rw-r--r-- 1 suc1 sfgi    0 Jan 29 12:53 cartilage_P18S_loc1.is

> Tue Jan 29 09:46:51 2019 Pre-process v.0.189 started
> Error in hist.default(data, xlim = c(0, 10000), breaks = 1000, xlab = "Insert size",  : 
>   'x' must be numeric
> Calls: hist -> hist.default
> Execution halted
> Tue Jan 29 12:54:04 2019 Pre-process finished
> Time used: 3:7:13

I tried to increase insertion size to 10000 bp, in case that no insertion is smaller than 1000 bp. It still failed. I used picard `CollectInsertSizeMetrics.jar` to calculate insert size. Here are the report

> ## METRICS CLASS        picard.analysis.InsertSizeMetrics
> MEDIAN_INSERT_SIZE      MEDIAN_ABSOLUTE_DEVIATION       MIN_INSERT_SIZE MAX_INSERT_SIZE MEAN_INSERT_SIZE        STANDARD_DEVIATION      READ_PAIRS      PAIR_ORIENTATION        WIDTH_OF_10_PERCENT     WIDTH_OF_20_PERCENT     WIDTH_OF_30_PERCENT     WIDTH_OF_40_PERCENT     WIDTH_OF_50_PERCENT     WIDTH_OF_60_PERCENT     WIDTH_OF_70_PERCENT     WIDTH_OF_80_PERCENT     WIDTH_OF_90_PERCENT     WIDTH_OF_99_PERCENT     SAMPLE  LIBRARY READ_GROUP
> 140     50      2       247030253       163.993181      103.729549      294915720       FR      21      39      59      81      101     125     153     193     373     1393

Thus insertion size is as expected --- majority pair insertion size falls into 1000bp range. 

From the perl script, `.is` file should be generated by `bamreader` command. However, when I run `bamreader` alone, it does not generate `.is` either. 

Could you please help?

**Comment by simoncchu** [comment]
Is R/3.4.1 or later? It seems the R script failed to run.

**Comment by sckinta** [comment]
I dont think it is R script problem. From the pre_process.pl perl code, R script is to plot the insertion size file. However, the insertion size file was not even generated before R script was performed. 

> ls -lhtr cartilage_P18S_loc1.srt.dupmark
> total 1.8G
> -rw-r--r-- 1 suc1 sfgi 679M Jan 29 12:53 cartilage_P18S_loc1_1.fq.gz
> -rw-r--r-- 1 suc1 sfgi 834M Jan 29 12:53 cartilage_P18S_loc1_2.fq.gz
> **-rw-r--r-- 1 suc1 sfgi 0 Jan 29 12:53 cartilage_P18S_loc1.is**

That is why R script failed to read any data in. the data table is null. 

I dont understand why the insertion size file fail to be generated.
